lib/ddtrace/configuration/components.rb

```diff
@@ -96,11 +109,64 @@ def build_tracer_options(settings)
             opts[:writer_options] = settings.writer_options if settings.writer.nil?
           end
         end
+
+        def build_profiler_recorder(settings)
+          event_classes = []
+
+          if settings.profiling.enabled && settings.profiling.cpu.enabled
+            event_classes << Datadog::Profiling::Events::StackSample
+          end
+
+          Datadog::Profiling::Recorder.new(event_classes, settings.profiling.max_events)
+        end
+
+        def build_profiler_collectors(settings, recorder)
+          return [] unless settings.profiling.enabled && settings.profiling.cpu.enabled
+
+          [
+            Datadog::Profiling::Collectors::Stack.new(
+              recorder,
+              enabled: settings.profiling.cpu.enabled,
+              max_frames: settings.profiling.cpu.max_frames,
+              max_time_usage_pct: settings.profiling.cpu.max_time_usage_pct
+              # TODO: Provide proc that identifies Datadog worker threads.
+              # ignore_thread: settings.profiling.ignore_profiler
+            )
+          ]
+        end
+
+        def build_profiler_exporters(settings)
+          if settings.profiling.exporter.instances.is_a?(Array)
+            settings.profiling.exporter.instances
+          else
+            transport = if settings.profiling.exporter.transport
+                          settings.profiling.exporter.transport
+                        else
+                          transport_options = settings.profiling.exporter.transport_options.dup
+                          transport_options[:site] ||= settings.site if settings.site
+                          transport_options[:api_key] ||= settings.api_key if settings.api_key
```

Comment:
That should not be necessary since we only support upload via the agent now.

Reply:
agreed for now agent mode only is fine but good for us to know agentless support is just a couple lines away